### DISCUSSION TITLE
Low brightness in beam visualization.

### DISCUSSION THREAD

**Discussion by waveguiding** [NanoComp/meep#2248] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/110013755/192198303-f6dce253-aa60-4c7e-b87f-bebddb18917b.png)

Never posted a question on github before, please forgive any question upload mistakes.

As seen in the image above, I'm having trouble visualizing beam propagation in a prism. The source is a GaussianBeamSource. The beam is not bright enough to see. I assume the issue is that normalization and spread of the beam, but I need to see how this beam propagates to create structures in its path. I thought about manipulating the amplitude but don't know how to modify that in this specific source type, or if it would be normalized back to the same state of brightness/visibility. Another guess is that the short wavelength (473nm) combined with the large sim size (50x50um) is leading to low visibility.

Thanks in advance if anyone can offer help or advice. Hope this group is still active.

**Comment by waveguiding** [comment]
Relevant code snippets:


# basic simulation parameters
resolution = 50  # pixels/um
cell = mp.Vector3(100, 100, 0)
pml_layers = [mp.PML(1.0)]

fcen = 1/0.473  # pulse center frequency
df = 0.1  # pulse width (in frequency)

sources = [mp.GaussianBeamSource(
                src = mp.ContinuousSource(wavelength = 0.473),
                center = mp.Vector3(-40,22.5,0),
                size = mp.Vector3(0,4.1,0),
                component = mp.Hz,
                beam_x0 = mp.Vector3(0,0,0),
                beam_kdir = mp.Vector3(1,0,0),
                beam_w0 = 2.05,
                beam_E0 = mp.Vector3(0,0,1)
                )]

plt.figure()
sim.plot2D(fields=mp.Ez, eps_parameters={'contour':True})
plt.axis('on')
plt.show()

**Comment by oskooi** [comment, marked as answer]
The problem is that you are using an `mp.Hz` source in a 2D simulation but plotting the `mp.Ez` fields which is zero by orthogonality. This is why your figure contains no fields. In your call to `plot2D`, replace `fields.mp.Ez` with `fields=mp.Hz`.

More generally, you can also try creating an animation of the time-dependent field propagation. There are a couple of ways to do this:
1. Write a [step function](https://meep.readthedocs.io/en/latest/Python_User_Interface/#output-functions) to output the fields (in this case, `mp.Hz` is the scalar field in the 2D simulation) to *disk* as an HDF5 file using `mp.output_hfield_z` at every Δt time units. Then, in post processing from the shell terminal, convert the HDF5 files to PNG via `$ h5topng -vZc bluered -C epsilon.h5 hz-000*.h5` where `epsilon.h5` is the ε structure (obtained using `mp.output_epsilon`) and `hz-000*.h5` are the snapshots of $H_z$ at various times. The `-C epsilon.h5` option is to superimpose the contours of the ε structure on the fields. To combine these PNG image files into an animated GIF, do e.g.: `$ convert -delay 20 hz-000*.png animation.gif`. For an example, see this [tutorial](https://meep.readthedocs.io/en/latest/Python_Tutorials/Basics/#a-90-bend).
2. Do the same thing as (1) in *memory* using [`mp.Animate2D`](https://meep.readthedocs.io/en/latest/Python_User_Interface/#animate2d).
